python_modules/dagster/dagster_tests/core_tests/execution_tests/test_context.py

```diff
@@ -286,6 +305,83 @@ def op_annotation(context: OpExecutionContext, *args):
     op_annotation_graph.to_job(name="op_annotation_job").execute_in_process()
 
 
+def test_context_provided_to_asset_check():
+    instance = DagsterInstance.ephemeral()
+
+    def execute_assets_and_checks(assets=None, asset_checks=None, raise_on_error: bool = True):
+        defs = Definitions(assets=assets, asset_checks=asset_checks)
+        job_def = defs.get_implicit_global_asset_job_def()
+        return job_def.execute_in_process(raise_on_error=raise_on_error, instance=instance)
+
+    @asset
+    def to_check():
+        return 1
+
+    @asset_check(asset=to_check)
+    def no_annotation(context):
+        assert isinstance(context, AssetExecutionContext)
+
+    execute_assets_and_checks(assets=[to_check], asset_checks=[no_annotation])
+
+    @asset_check(asset=to_check)
+    def asset_annotation(context: AssetExecutionContext):
+        assert isinstance(context, AssetExecutionContext)
+
+    execute_assets_and_checks(assets=[to_check], asset_checks=[asset_annotation])
+
+    @asset_check(asset=to_check)
+    def op_annotation(context: OpExecutionContext):
+        assert isinstance(context, OpExecutionContext)
+        # AssetExecutionContext is an instance of OpExecutionContext, so add this additional check
+        assert not isinstance(context, AssetExecutionContext)
+
+    execute_assets_and_checks(assets=[to_check], asset_checks=[op_annotation])
+
+
+def test_context_provided_to_blocking_asset_check():
+    instance = DagsterInstance.ephemeral()
+
+    def execute_assets_and_checks(assets=None, asset_checks=None, raise_on_error: bool = True):
+        defs = Definitions(assets=assets, asset_checks=asset_checks)
+        job_def = defs.get_implicit_global_asset_job_def()
+        return job_def.execute_in_process(raise_on_error=raise_on_error, instance=instance)
+
+    @asset
+    def to_check():
+        return 1
+
+    # @asset_check(asset=to_check)
+    # def no_annotation(context):
```

Comment:
still to do - when used as part of a blocking asset, we lose the ability to see if the op of an asset check is an asset check, so we don't supply the correct context type. need to investigate and fix